plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/commands/ToggleConnectionBendpointCommand.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.commands;
+
+import org.eclipse.fordiac.ide.model.libraryElement.Connection;
+import org.eclipse.fordiac.ide.model.libraryElement.ConnectionRoutingData;
+import org.eclipse.gef.commands.Command;
+
+public class ToggleConnectionBendpointCommand extends Command {
+	private final org.eclipse.draw2d.Connection figure;
+	private final ConnectionRoutingData connectionRoutingData;
+
+	public ToggleConnectionBendpointCommand(final Connection connection, final org.eclipse.draw2d.Connection figure) {
+		this.connectionRoutingData = connection.getRoutingData();
+		this.figure = figure;
+	}
+
+	@Override
+	public void execute() {
+		toggle5Segments();
```

Comment:
As you are not storing the original connection routing data undo is not working. Also I would split it into two commands:
  - ConvertToThreeSegmentsCon
  - ConvertToFiveSegmentsCon
  
  Most probably with a common base class with the common functionality. 
  The advantage is that you could in both commands create a new connectionrouting data suitable for the different kind of connection.